sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,92 +16,267 @@
 
 package com.nvidia.spark.rapids
 
+import java.io.FileNotFoundException
+
+import scala.io.Source
+import scala.sys.process.{Process, ProcessLogger}
+
 import org.apache.hadoop.fs.Path
 
+import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.{Expression, PlanExpression}
 import org.apache.spark.sql.execution.datasources.{FileIndex, HadoopFsRelation, InMemoryFileIndex}
 import org.apache.spark.sql.execution.datasources.rapids.GpuPartitioningUtils
 
-object AlluxioUtils {
+object AlluxioUtils extends Logging {
+  val mountedBuckets: scala.collection.mutable.Map[String, String] = scala.collection.mutable.Map()
+  var alluxioCmd: Seq[String] = null
+  var alluxioMasterHost: Option[String] = None
+  var alluxio_home: String = "/opt/alluxio-2.8.0"
+  var isInit: Boolean = false
+
+  // Read out alluxio.master.hostname, alluxio.master.rpc.port
+  // from Alluxio's conf alluxio-site.properties
+  // We require an environment variable "ALLUXIO_HOME"
+  // This function will only read once from ALLUXIO/conf.
+  private def initAlluxioInfo(conf: RapidsConf): Unit = {
+    this.synchronized {
+      alluxio_home = scala.util.Properties.envOrElse("ALLUXIO_HOME", "/opt/alluxio-2.8.0")
+      alluxioCmd = conf.getAlluxioCmd.getOrElse(
+        Seq("su", "ubuntu", "-c", s"$alluxio_home/bin/alluxio"))
+
+      if (!isInit) {
+        // Default to read from /opt/alluxio-2.8.0 if not setting ALLUXIO_HOME
+        var alluxio_port: String = "19998"
+        var alluxio_master: String = null
+        var buffered_source: Source = null
+        try {
+          buffered_source = Source.fromFile(alluxio_home + "/conf/alluxio-site.properties")
+          for (line <- buffered_source.getLines) {
+            if (line.startsWith("alluxio.master.hostname")) {
+              alluxio_master = line.split('=')(1).trim
+            } else if (line.startsWith("alluxio.master.rpc.port")) {
```

Comment:
same as above

Reply:
Use java.util.Properties to load the alluxio-site.properties to avoid space issue.